test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommand_AlgorithmEquivalencyTests.cs

```diff
@@ -19,6 +19,91 @@ namespace NuGet.Commands.FuncTest
 {
     public partial class RestoreCommandTests
     {
+        [Fact]
+        // Project1 -> A 1.0.*  -> C 1.0.0
+        //             B 1.0.*  -> C 1.0.0
+        //             Project2 -> C 1.0.*
+        public async Task RestoreCommand_WithTransitiveFloatingVersion_ResolvesCorrectly()
```

Comment:
Do we also need tests for when the requested range is `1.*`, and `*`?

Reply:
In debugging this, I don't think it had a functional difference, but it wouldn't hurt to add the test coverage just in case.